### ISSUE TITLE
mysqli: regression on errorMsg()/errorCode()

### ISSUE THREAD

**Issue by Gabriel-Grosu** [ADOdb/ADOdb#872]
I see a difference in how errors are reported for duplicate entries in mysqli (error number 1062). Up to version (including) 5.21.4  all duplicate entries outputted an error (ErrorNo() and ErrorMsg() are correctly populated. For newer versions (5.22.*) duplicate entries are not outputted as errors anymore (ErrorNo() is 0 and ErrorMsg() is "").

**Comment by dregad** [comment]
It works for me...
```php
$db = NewADOConnection('mysqli');
$db->connect($host, $user, $password, $database);
// CREATE TABLE test (id INT, PRIMARY KEY (id));
$db->execute("insert into test (id) values (1)");
echo $db->ErrorNo() . " - " . $db->ErrorMsg() . "\n";
```

I get the same output `1062 - Duplicate entry '1' for key 'test.PRIMARY'`, as expected, for both 5.21.4 and 5.22.3.

Please provide detailed steps to reproduce.

**Comment by Gabriel-Grosu** [comment]
I am also unable to reproduce it now, I have tried several ways and different environments. I will close the issue now and come back if I run into this again. Sorry for the "false alarm".

**Comment by Gabriel-Grosu** [comment]
I am also unable to reproduce it now, I have tried several ways and different environments. I will close the issue now and come back if I run into this again. Sorry for the "false alarm".

**Comment by dregad** [comment]
Thanks for the update and code snippet, I can reproduce it now.

I ran a quick git bisect, which tells me the regression was introduced by commit a4b5110cf0740225caa345e969b1fcfefe7000a2 (see #655). Further analysis is needed to find out what is causing the problem.

**Comment by dregad** [comment]
@Gabriel-Grosu any feedback on proposed fix #876 ?
